source/code/plugins/out_oms_diag.rb

```diff
@@ -0,0 +1,112 @@
+module Fluent
+
+  class OutputOMSDiag < BufferedOutput
+
+    Plugin.register_output('out_oms_diag', self)
+
+    def initialize
+      super
+      require 'net/http'
+      require 'net/https'
+      require 'uri'
+      require_relative 'omslog'
+      require_relative 'oms_configuration'
+      require_relative 'oms_common'
+      require_relative 'oms_diag'
+    end
+
+    config_param :omsadmin_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/conf/omsadmin.conf'
+    config_param :cert_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.crt'
+    config_param :key_path, :string, :default => '/etc/opt/microsoft/omsagent/certs/oms.key'
+    config_param :proxy_conf_path, :string, :default => '/etc/opt/microsoft/omsagent/proxy.conf'
+    # TODO: To make compress default to true once supported
```

Comment:
It would be possible to make a DSC module to update omsagent.conf; see [nxOMSPerfCounter](https://github.com/Microsoft/PowerShell-DSC-for-Linux/blob/master/Providers/Scripts/2.4x-2.5x/Scripts/nxOMSPerfCounter.py#L175) and how it updates perfcounter sources there.